sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala

```diff
@@ -57,14 +57,14 @@ case class Average(child: Expression) extends DeclarativeAggregate with Implicit
   private lazy val resultType = child.dataType match {
     case DecimalType.Fixed(p, s) =>
       DecimalType.bounded(p + 4, s + 4)
-    case _: YearMonthIntervalType => YearMonthIntervalType
+    case _: YearMonthIntervalType => YearMonthIntervalType()
```

Comment:
@MaxGekk For `Sum`,  `resultType` is the same as `child.dataType` but for `Average` `resultType` is always `YearMonthIntervalType(0, 1)`. Is this intended behavior?